src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -1314,8 +1337,1089 @@ public boolean mapSwitchControlValue(String value) {
+		addAdvanceControlProperties(advancedControllableProperties,
+				createText(stats, streamGroup + StreamControllingMetric.STREAM_NAME.getName(), streamName));
+		// Populate Encapsulation dropdown control
+		addAdvanceControlProperties(advancedControllableProperties, createDropdown(stats, streamGroup + StreamControllingMetric.ENCAPSULATION.getName(), encapsulationList, encapsulation.getName()));
+
+		// Populate delete button
+		addAdvanceControlProperties(advancedControllableProperties, createButton(streamGroup + DecoderConstant.DELETE, DecoderConstant.DELETE, DecoderConstant.DELETING));
+
+		switch (encapsulation) {
+			case TS_OVER_UDP:
+				populateStreamControlCaseTSOverUDP(stats, advancedControllableProperties, streamInfo, streamGroup);
+				break;
+			case TS_OVER_RTP:
+				populateStreamControlCaseTSOverRTP(stats, advancedControllableProperties, streamInfo, streamGroup);
+				break;
+			case TS_OVER_SRT:
+				populateStreamControlCaseTSOverSRT(stats, advancedControllableProperties, streamInfo, streamGroup);
+		}
+
+		// Populate apply change and cancel button
+		populateApplyChangeAndCancelButtonForStream(stats, advancedControllableProperties, streamInfo.getId());
+	}
+
+	/**
+	 * This method is used for populate all create stream control properties:
+	 * <li>Protocol: TS over UDP</li>
+	 * <li>Protocol: TS over RTP</li>
+	 * <li>Protocol: TS over SRT</li>
+	 *
+	 * @param stats is the map that store all statistics
+	 * @param advancedControllableProperties is the list that store all controllable properties
+	 * @param streamInfo stream config info
+	 */
+	private void populateCreateStreamControl(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties, StreamInfo streamInfo, String streamGroup) {
+		// Get controllable property current value
+		String streamName = streamInfo.getName();
+		Encapsulation encapsulation = streamInfo.getEncapsulation();
+		List<String> encapsulationList = Encapsulation.getEncapsulationList();
+
+		// Populate stream name text control
+		advancedControllableProperties.add(
+				createText(stats, streamGroup + StreamControllingMetric.STREAM_NAME.getName(), streamName));
+
+		// Populate Encapsulation dropdown control
+		advancedControllableProperties.add(createDropdown(stats, streamGroup + StreamControllingMetric.ENCAPSULATION.getName(), encapsulationList, encapsulation.getName()));
+
+		// Populate create button
+		advancedControllableProperties.add(createButton(streamGroup + DecoderConstant.CREATE, DecoderConstant.CREATE, DecoderConstant.CREATING));
+		stats.put(streamGroup + DecoderConstant.CREATE, DecoderConstant.EMPTY);
+
+		switch (encapsulation) {
+			case TS_OVER_UDP:
+				populateStreamControlCaseTSOverUDP(stats, advancedControllableProperties, streamInfo, streamGroup);
+				break;
+			case TS_OVER_RTP:
+				populateStreamControlCaseTSOverRTP(stats, advancedControllableProperties, streamInfo, streamGroup);
+				break;
+			case TS_OVER_SRT:
```

Comment:
doesn't it need a default state, just in case? With a debug log entry, for instance, just to keep track of everything happening with the logic segment

Reply:
the encapsulation always returns the default value is TS over UDP if there is no value here, so I don't put the default statement here. But you are right, It better than if I have the default state here. I fixed it.